dbms/src/Common/ColumnsHashingImpl.h

```diff
@@ -59,24 +59,24 @@ struct LastElementCache<Data, false>
 template <typename Mapped>
 class EmplaceResultImpl
 {
-    Mapped & value;
-    Mapped & cached_value;
-    bool inserted;
+    Mapped * value = nullptr;
+    Mapped * cached_value = nullptr;
+    bool inserted = false;
```

Comment:
What specific issue does this change address?

Reply:
emplace_result_holder (std::optional<EmplaceResult>) needs copy assign operator